### DISCUSSION TITLE
OHLCV weird exception

### DISCUSSION THREAD

**Discussion by ToposFinance** [bmoscon/cryptofeed#913] [Q&A] [answered]
Hi,

Tried running your example of OHLCV on BTC-PERP

got 

cryptofeed.exceptions.UnsupportedSymbol: BTC-PERP is not supported on FTX

but certainly with the websocket I can stream this.

**Comment by bmoscon** [comment]
can you be more specific? The ohlcv example has coinbase in it, not FTX

**Reply by ToposFinance** [reply]
'''
Copyright (C) 2017-2022 Bryant Moscon - bmoscon@gmail.com
Please see the LICENSE file for the terms and conditions
associated with this software.
'''
from cryptofeed import FeedHandler
from cryptofeed.backends.aggregate import OHLCV
from cryptofeed.defines import TRADES
from cryptofeed.exchanges import FTX


async def ohlcv(data):
    print(data)


def main():
    f = FeedHandler()
    f.add_feed(FTX(symbols=['BTC-PERP'], channels=[TRADES], callbacks={TRADES: OHLCV(ohlcv, window=10)}))

    f.run()


if __name__ == '__main__':
    main()

**Reply by ToposFinance** [reply]
response
/Users/marcnunes/PycharmProjects/DeepPounder/venv/bin/python /Users/marcnunes/PycharmProjects/DeepPounder/main.py 
Traceback (most recent call last):
  File "/Users/marcnunes/PycharmProjects/DeepPounder/venv/lib/python3.8/site-packages/cryptofeed/exchange.py", line 143, in std_symbol_to_exchange_symbol
    return self.normalized_symbol_mapping[symbol]
KeyError: 'BTC-PERP'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/marcnunes/PycharmProjects/DeepPounder/main.py", line 24, in <module>
    main()
  File "/Users/marcnunes/PycharmProjects/DeepPounder/main.py", line 18, in main
    f.add_feed(FTX(symbols=['BTC-PERP'], channels=[TRADES], callbacks={TRADES: OHLCV(ohlcv, window=10)}))
  File "/Users/marcnunes/PycharmProjects/DeepPounder/venv/lib/python3.8/site-packages/cryptofeed/feed.py", line 117, in __init__
    symbols = [self.std_symbol_to_exchange_symbol(symbol) for symbol in symbols]
  File "/Users/marcnunes/PycharmProjects/DeepPounder/venv/lib/python3.8/site-packages/cryptofeed/feed.py", line 117, in <listcomp>
    symbols = [self.std_symbol_to_exchange_symbol(symbol) for symbol in symbols]
  File "/Users/marcnunes/PycharmProjects/DeepPounder/venv/lib/python3.8/site-packages/cryptofeed/exchange.py", line 148, in std_symbol_to_exchange_symbol
    raise UnsupportedSymbol(f'{symbol} is not supported on {self.id}')
cryptofeed.exceptions.UnsupportedSymbol: BTC-PERP is not supported on FTX

Process finished with exit code 1

**Reply by ToposFinance** [reply]
but it's very easy to use websockets to stream BTC-PERP with ftx endpoint

**Reply by ToposFinance** [reply]
on the other hand

**Reply by ToposFinance** [reply]
'''
Copyright (C) 2017-2022 Bryant Moscon - bmoscon@gmail.com
Please see the LICENSE file for the terms and conditions
associated with this software.
'''
from cryptofeed import FeedHandler
from cryptofeed.backends.aggregate import OHLCV
from cryptofeed.defines import TRADES
from cryptofeed.exchanges import FTX


async def ohlcv(data):
    print(data)


def main():
    f = FeedHandler()
    f.add_feed(FTX(symbols=['BTC-USD'], channels=[TRADES], callbacks={TRADES: OHLCV(ohlcv, window=10)}))

    f.run()


if __name__ == '__main__':
    main()

**Reply by ToposFinance** [reply]
/Users/marcnunes/PycharmProjects/DeepPounder/venv/bin/python /Users/marcnunes/PycharmProjects/DeepPounder/main.py 
{'BTC-USD': {'open': Decimal('19833.0'), 'high': Decimal('19856.0'), 'low': Decimal('19833.0'), 'close': Decimal('19843.0'), 'volume': Decimal('34.1680'), 'vwap': Decimal('19841.32192694919222664481386')}}
{'BTC-USD': {'open': Decimal('19841.0'), 'high': Decimal('19851.0'), 'low': Decimal('19841.0'), 'close': Decimal('19847.0'), 'volume': Decimal('4.9620'), 'vwap': Decimal('19848.23264812575574365175333')}}

Process finished with exit code 0

**Reply by ToposFinance** [reply]
works

**Reply by ToposFinance** [reply]
COINBASE works

**Reply by ToposFinance** [reply]
Figured out the problem. Will post it tomorrow morning  Hong Kong time

**Comment by ToposFinance** [comment, marked as answer]
self.normalized_symbol_mapping in exchange.py contains a dictionary whose keys map to exchange symbols. so, the key that maps to 'BTC-PERP' on FTX is 'BTC-USD-PERP'

**Comment by bmoscon** [comment]
@ToposFinance I think you were using an old version of the code - that sample/demo code you posted does not exist in the repo currently.
